### PR TITLE
[#87155688][1pts] Filters / Hide option for Published Blog posts.

### PR DESCRIPTION
**Pull request by zalapeach**
## This pull request involves

* Adds some filters for blog admin
* Adds cucumbers tests

### CODE REVIEW

**Comment by heridev** [comment]
LGTM

File: app/views/crowdblog/admin/posts/index.html.slim [hunk 1]
```diff
@@ -8,12 +8,14 @@
       ul.nav.nav-pills
         li class="#{@state ? '' : 'active'}"
           = link_to 'All', crowdblog.admin_posts_path
+        li class="#{@state == 'drafted' ? 'active' : ''}"
```
Comment (mumoc) [inline app/views/crowdblog/admin/posts/index.html.slim]: you could move this comparison to helper ... or javascript function or somewhere else, just to clean the view ^^

File: app/views/crowdblog/admin/posts/index.html.slim [hunk 2]
```diff
@@ -8,12 +8,14 @@
...
+          = link_to 'Drafted', crowdblog.admin_posts_by_state_path('drafted')
         li class="#{@state == 'finished' ? 'active' : ''}"
           = link_to 'Finished', main_app.admin_finished_posts_path('finished')
+        li class="#{@state == 'reviewed' ? 'active' : ''}"
+          = link_to 'Reviewed', main_app.admin_finished_posts_path('reviewed')
```
Comment (mumoc) [inline app/views/crowdblog/admin/posts/index.html.slim]: why do you have 2 different types of named routes? ```admin_posts_by_state_path(state)``` and ```admin_finished_posts_path``` shouldn't you need to use the first one? ```admin_posts_by_state_pateh('reviewed')``` or change it to say ```admin_darfted_posts_path``` ....

Please check the consistency of your named routes ^^

File: config/initializers/posts_controller_decorator.rb [hunk 3]
```diff
@@ -1,6 +1,13 @@
 Crowdblog::Admin::PostsController.class_eval do
   before_action :load_categories, only: [:edit]
 
+  def index
+    @state = params[:state]
```
Comment (mumoc) [inline config/initializers/posts_controller_decorator.rb]: I'm not sure if the @state variable is needed, I would move it to a helper method in the controller ... or even better, you could work with the Post model and create a scope where you could just pass the state and if none is passed to return all the published_at ...

File: features/admin_filters.feature [hunk 4]
```diff
@@ -0,0 +1,39 @@
+@omniauth_test
+Feature: Filters for admin
+
+  Background: Admin checking different filters
+    Given I am logged in as an admin user
+      And the following posts by "foo":
+        | title                | body                             | state     | published |
+        | Wecking Crew         | I can't believe it, is he mario? | drafted   | false     |
+        | Xenoblade Chronicles | This is the monado's power!      | finished  | false     |
+        | Bayonetta 2          | Do you want to touch me?         | reviewed  | false     |
```
Comment (mumoc) [inline features/admin_filters.feature]: I see what you did here....

File: config/initializers/posts_controller_decorator.rb (hunk 3 above)
Comment (zalapeach) [inline config/initializers/posts_controller_decorator.rb]: the @state variable is needed in the view in order to know what state of the blogpost is need

**Comment by zalapeach** [comment]
Done @mumoc

**Pull request merged**